lib/pbench/agent/tool_meister.py

```diff
@@ -1285,6 +1285,50 @@ def stop_tools(self, data):
             self._send_client_status("success")
         return failures
 
+    def _create_tar(
+        self,
+        directory: Path,
+        tar_file: Path,
+        tar_args: List = [],
+        retry: bool = False,
+    ) -> subprocess.CompletedProcess:
+        """
+        Creates a tar file at a given tar file path.
+        If an error occurs it will retry by suppressing all the warnings.
+        """
+        if not tar_args:
+            tar_args = [
+                self.tar_path,
+                "--create",
+                "--xz",
+                "--force-local",
+                f"--file={tar_file}",
+                directory.name,
+            ]
+
+        # Invoke tar directly for efficiency.
+        cp = subprocess.run(
+            tar_args,
+            cwd=directory.parent,
+            stdin=None,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        if cp.returncode != 0:
+            self.logger.warning(
+                "Tar ball creation failed with '%s', %s; returncode: %d",
+                cp.stdout.decode("utf-8"),
+                "re-trying ignoring all warnings using --warning=none" if retry else "",
+                cp.returncode,
+            )
+            if retry:
+                tar_args.insert(2, "--warning=none")
+                self._create_tar(directory, tar_file, tar_args)
+            else:
+                return cp
+        else:
+            return cp
```

Comment:
_[Hopefully you will restructure this code and this comment will be moot, but....]_

In the event that we make the recursive call to `_create_tar()` at line 1326, I think we'll return `None` instead of a `CompletedProcess` object.

In any case, I don't think we need multiple `return` statements, here -- I think you can remove all the `else`'s and just return `cp` at the end (assuming that we set it properly...).